backend/src/modules/product/product.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Model } from 'mongoose';
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Product } from './product.interface';
+import { CreateProductDto } from './dto/create-product.dto';
+
+@Injectable()
+export class ProductService {
+  constructor(@InjectModel('Product') private readonly productModel: Model<Product>) {}
+
+  async create(createCatDto: CreateProductDto): Promise<Product> {
+    const createdCat = new this.productModel(createCatDto);
+    return await createdCat.save();
+  }
+
+  async findAll(skip: number = 0, limit: number = 4): Promise<Product[]> {
+    return await this.productModel.find(
+      null,
+      null,
+      { skip: skip, limit: limit }
```

Comment:
Can be wrote simpler - { skip, limit }.
And it will be more readable, if it is moved into const above function call